diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f41f7a..b244419 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,10 @@ macro(build_sol)
 
   include(ExternalProject)
   externalproject_add(sol-ext
-    GIT_REPOSITORY https://github.com/ThePhD/sol2.git
-    GIT_TAG main
+    URL ${CMAKE_CURRENT_SOURCE_DIR}/sol2.tar.gz
     TIMEOUT 6000
+    PATCH_COMMAND patch -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/0-sol2-change-cmake-download.patch
+    COMMAND mv ${CMAKE_CURRENT_SOURCE_DIR}/lua-5.4.4.tar.gz .
     ${cmake_commands}
     CMAKE_ARGS
       ${cmake_configure_args}
-- 
2.41.0
