diff -Naur ros-humble-ignition-math6-vendor-0.0.2_org/CMakeLists.txt ros-humble-ignition-math6-vendor-0.0.2/CMakeLists.txt
--- ros-humble-ignition-math6-vendor-0.0.2_org/CMakeLists.txt	2023-04-30 22:13:16.830547196 +0800
+++ ros-humble-ignition-math6-vendor-0.0.2/CMakeLists.txt	2023-04-30 22:13:46.906092327 +0800
@@ -51,9 +51,7 @@
   include(ExternalProject)
 
   externalproject_add(ignition-math6-${IGNITION_MATH6_TARGET_VERSION}
-    GIT_REPOSITORY https://github.com/ignitionrobotics/ign-math.git
-    GIT_TAG ignition-math6_${IGNITION_MATH6_TARGET_VERSION}
-    GIT_CONFIG advice.detachedHead=false
+    URL ${CMAKE_CURRENT_SOURCE_DIR}/ignition-math6_6.9.2.tar.gz
     # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
     # See https://github.com/ament/uncrustify_vendor/pull/22 for details
     UPDATE_COMMAND ""
