diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54a6521..86884ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,8 +7,7 @@ ament_auto_find_build_dependencies()
 
 include(ExternalProject)
 ExternalProject_Add(qpoases_3_2
-  SVN_REPOSITORY https://projects.coin-or.org/svn/qpOASES/stable/3.2
-  SVN_TRUST_CERT TRUE
+  URL ${CMAKE_CURRENT_SOURCE_DIR}/qpoases_3_2/3.2.0.tar.gz
   BUILD_IN_SOURCE TRUE
   INSTALL_COMMAND ""
 )
